pybop/costs/base_cost.py

```diff
@@ -58,6 +58,7 @@ def __call__(
         inputs: Union[Inputs, list],
         calculate_grad: bool = False,
         apply_transform: bool = False,
+        minimising: bool = True,
```

Comment:
If a user wants to use the cost outside of the optimisation, `minimising` doesn't convey what we are doing behind the scenes. I think keeping `minimising` within the optimisers and using it to set the `negate_output` conveys more to the user. I'm not tied to `negate_output`, perhaps `flip_sign` or similar would get us on the same page? Otherwise, maybe @martinjrobins can weigh in so we can get a quorum.